src/base/bittorrent/sessionimpl.cpp

```diff
@@ -5967,11 +5977,19 @@ void SessionImpl::handleExternalIPAlert(const lt::external_ip_alert *alert)
     LogMsg(tr("Detected external IP. IP: \"%1\"")
         .arg(externalIP), Log::INFO);
 
-    if (m_lastExternalIP != externalIP)
+    const bool isIPv6 = alert->external_address.is_v6();
+    const bool isIPv4 = alert->external_address.is_v4();
+    if (isIPv6 && (externalIP != m_lastExternalIPv6Address))
+    {
+        if (isReannounceWhenAddressChangedEnabled() && !m_lastExternalIPv6Address.isEmpty())
+            reannounceToAllTrackers();
```

Comment:
OFFTOPIC
This logic should be reconsidered further. I feel it incorrect.

Reply:
Using VPN external ipv6 address can change very often, so there may be too many reannounces.

Reply:
For an example (these are Cloudflare IP)
![2024-11-03_114002](https://github.com/user-attachments/assets/98670749-c273-4fbc-a81c-37e551c0cbd5)